src/Pages/MembershipApplication/MembershipPlan.js

```diff
@@ -10,83 +10,76 @@ import spring from './assets/spring.jpg'
 import spring2 from './assets/spring2.jpeg'
 import { Container, Button, Row } from 'reactstrap'
 import { memberApplicationState, membershipPlans } from '../../Enums'
+import { getSemesterPlan, getYearPlan } from './GetPlans'
 
 class MembershipPlan extends Component {
   constructor (props) {
     super(props)
     this.state = {
-      img1: winter,
-      img2: winter2,
+      yearPicture: winter2,
+      semesterPicture: winter,
       activeId: undefined,
-      planSelected: false
+      planSelected: false,
+      yearPlan: getYearPlan(),
+      year: new Date().getFullYear(),
+      semesterPlan: getSemesterPlan(),
+      planType: []
     }
   }
 
-  cardSelected (id) {
-    if (id === 'Spring and Fall 2020') {
+  cardSelected (plan) {
+    if (
+      plan.includes('Spring and Fall') ||
+      plan.includes('Fall ' + this.state.year + ' and')
+    ) {
       this.props.setSelectedPlan(membershipPlans.YEAR)
-    } else if (id === 'Spring 2020') {
+    } else {
       this.props.setSelectedPlan(membershipPlans.SEMESTER)
     }
     this.setState({
-      activeId: id,
+      activeId: plan,
       planSelected: true
     })
   }
 
-  getBody (title) {
-    if (title === 'Spring and Fall 2020') {
-      return (
-        <h6 style={{ margin: 5 }}>
-          Access to our club room during legal building hours. Access to our
-          events and news. And much more! This membership lasts a full year and
-          expires on December 20th, 2020. Sign Up @ SCE (ENGR 294){' '}
-        </h6>
-      )
+  getExpirationDate (plan) {
+    if (
+      plan.includes('Fall ' + this.state.year) ||
+      plan.includes('Spring and Fall ' + this.state.year)
+    ) {
+      return `May 20th, ${this.state.year}`
     }
-    return (
-      <h6 style={{ margin: 5 }}>
-        Access to our club room during legal building hours. Access to our
-        events and news. And much more! This membership lasts a semester and
-        expires on May 20th, 2020. Sign Up @ SCE (ENGR 294)
-      </h6>
-    )
+    return `December 20th, ${this.state.year}`
   }
 
   componentDidMount () {
     this.changeSeason()
-  }
-
-  makeCard (title, img) {
-    return (
-      <div
-        className={
-          title === this.state.activeId
-            ? 'card membership-card active-plan'
-            : 'card membership-card'
+    this.setState({
```

Comment:
move this to the constructor. there is no need to set this later as it is constant.

Reply:
It needs to be after changeSeason()
changeSeason() changes the pictures then we proceed to set planType